dcm4che-tool/dcm4che-tool-findscu/src/main/java/org/dcm4che3/tool/findscu/FindSCU.java

```diff
@@ -437,12 +443,30 @@ private static void configureCancel(FindSCU main, CommandLine cl) {
     }
 
     private static void configureKeys(FindSCU main, CommandLine cl) {
-        CLIUtils.addEmptyAttributes(main.keys, cl.getOptionValues("r"));
-        CLIUtils.addAttributes(main.keys, cl.getOptionValues("m"));
+        addFilesAsArgs(main, cl);
+        CLIUtils.addEmptyAttributes(main.keys, main.returnKeys.toArray(new String[main.returnKeys.size()]));
+        CLIUtils.addAttributes(main.keys, main.matchingKeys.toArray(new String[main.matchingKeys.size()]));
         if (cl.hasOption("L"))
             main.addLevel(cl.getOptionValue("L"));
         if (cl.hasOption("i"))
-            main.setInputFilter(CLIUtils.toTags(cl.getOptionValues("i")));
+            main.setInputFilter(CLIUtils.toTags(main.iKeys.toArray(new String[main.iKeys.size()])));
+    }
+
+    private static void addFilesAsArgs(FindSCU main, CommandLine cl) {
+        addFilesAsArgs(main.argList, cl.getOptionValues("r"), main.returnKeys);
+        addFilesAsArgs(main.argList, cl.getOptionValues("m"), main.matchingKeys);
+        addFilesAsArgs(main.argList, cl.getOptionValues("i"), main.iKeys);
+    }
+
+    private static void addFilesAsArgs(List<String> argList, String[] optVals, List<String> keys) {
+        if (optVals != null)
+            for (String key : optVals) {
+                File f = new File(key);
+                if (f.exists())
```

Comment:
Ah, I think I understand the problem now.
Maybe we should create these options with
```
.hasArg()
```
instead of
```
.hasArgs()
```
to avoid this ambigouity?

Just imagine the confusion should someone by conincidence have a file called e.g. "PatientName" in his working directory. Your current implementation is unpredictable.